Update OpenAPI overview (#257)
SAP/cloud-sdk: docs/js/features/openapi/overview.mdx

---
id: overview

> title: Overview

Reviewer: The title of the odata overview is `OData`, should we use OpenAPI here?

> title: Overview

Author: ```suggestion
title: OpenAPI Overview
```

> hide_title: false

Reviewer: Please correct me, it seems, this value by default is false.

> hide_table_of_contents: false

Reviewer: Also, this is not needed.

> hide_table_of_contents: false

Author: ```suggestion
```

sidebar_label: Overview

> description: Type-safe client generators for OpenAPI services for Typescript and JavaScript

Reviewer: ```suggestion
description: Type-safe client generators for OpenAPI services for TypeScript and JavaScript
```

> keywords:

Reviewer: question: should we align with the odata overview about the keywords?

Author: not really, they doesn't have much weight for Google.

  - sap
  - cloud
  - sdk
  - cloud native
  - cloud sdk
  - sap cloud sdk
  - openapi
  - JavaScript

>   - Typescript

Reviewer: ```suggestion
  - TypeScript
```

  - OpenAPI
---

:::tip Released as generally available

> OpenAPI code generator and type-safe client are released as generally available and ready for productive use.

Reviewer: ```suggestion
OpenAPI code generator and type-safe client are released as General Availability and ready for productive use.
```

:::

> ## OpenAPI Based Type-Safe Client for RESTful Services in Typescript and JavaScript

Reviewer: ```suggestion
## OpenAPI Based Type-Safe Client for RESTful Services in TypeScript and JavaScript
```

[OpenAPI](https://www.openapis.org/) standard is a popular format to define RESTful APIs.
Together with [OData](../odata/overview.mdx) it is one of the key API definition formats used at SAP.
The SAP Cloud SDK provides convenient tooling to make the consumption of OpenAPI services easy and convenient.
For that we developed an [OpenAPI code generator](generate-openapi-client.mdx) from the ground up.

> It's a command line tool capable of converting any OpenAPI specification into a type-safe Typescript or JavaScript client library that is nicely integrated with other SAP Cloud SDK feature like [connectivity](../connectivity/destination.mdx) and other abstractions for the SAP Business Technology Platform.

Reviewer: ```suggestion
It's a command line tool capable of converting any OpenAPI specification into a type-safe TypeScript or JavaScript client library that is nicely integrated with other SAP Cloud SDK features like [connectivity](../connectivity/destination.mdx) and other abstractions for the SAP Business Technology Platform.
```

> You can use the OpenAPI generator to create a client for any OpenAPI-based API service.

Reviewer: This might be wrong, because we only support version 2 and 3.

If you encounter any problems or errors with OpenAPI tooling give us feedback via our [support channels](../../../overview/support.mdx).

### Pregenerated Type-Safe OpenAPI Client

> To speed up things for developers and reduce maintenance, we are working with different SAP services to release pre-generated client libraries available on [npm](https://www.npmjs.com/search?q=%40sap%2Fcloud-sdk-vdm).

Author: ```suggestion
We're working with service publishers on releasing OpenAPI client libraries. Stay tuned and follow our release notes for more details.
```

> To speed up things for developers and reduce maintenance, we are working with different SAP services to release pre-generated client libraries available on [NPM](https://www.npmjs.com/search?q=%40sap%2Fcloud-sdk-vdm).

Author: ```suggestion
To speed up things for developers and reduce maintenance, we are working with different SAP services to release pre-generated client libraries available on [npm](https://www.npmjs.com/search?q=%40sap%2Fcloud-sdk-vdm).
```

> To speed up things for developers and reduce maintenance, we are working with different SAP services to release pre-generated client libraries available on [npm](https://www.npmjs.com/search?q=%40sap%2Fcloud-sdk-vdm).

Reviewer: I'm afraid you won't find any regenerated OpenAPI clients for now.
Even until the end of the quarter, there might be only 1 or 2, and this link gives hundreds of vdms which are irrelavant.

Another way to get a client library is to [generate it yourself](generate-openapi-client.mdx) based on API specifications available on the [SAP API Business Hub](https://api.sap.com/).

> ## Why Using SAP Cloud SDK for Java With OpenAPI Services?

Reviewer: ```suggestion
## Why Using SAP Cloud SDK for JavaScript With OpenAPI Services?
```

> 

Reviewer: Actually, I realised, this section (or might be some other sections) are almost the same as the java one, when I saw the copy/paste error above.
We can make a reusable code and inject it in both java/js overview.

Author: I'll mind it for refactoring. For now, Java doesn't use components much. When we re-work our value prop will be a good time to break it into reusable pieces.

- You'll benefit from type-safety, less code boilerplate, and better developer experience.
- We take care of various complexities around the development of applications and extensions on the SAP Business Technology Platform.

> - You'll get convenience abstractions for Destinations, Authentication, Service bindings, CSRF and ETag tokens, Headers, and much more.

Reviewer: ```suggestion
- You'll get convenient abstractions for Destinations, Authentication, Service bindings, CSRF and ETag tokens, Headers, and much more.
```
With the reusable code mentioned above, you don't have to fix this for multiple times.

- We hide the complexities of cloud development making many tasks ridiculously easy.

> - You're getting best in class [support from the SAP Cloud SDK](https://github.com/SAP/cloud-sdk-js/issues/new/choose) development team

Reviewer: ```suggestion
- You're getting best in class [support from the SAP Cloud SDK](https://github.com/SAP/cloud-sdk-js/issues/new/choose) development team.
```
